core/chains/evm/logpoller/helper_test.go

```diff
@@ -47,7 +47,8 @@ func SetupTH(t *testing.T) TestHarness {
 			Balance: big.NewInt(0).Mul(big.NewInt(10), big.NewInt(1e18)),
 		},
 	}, 10e6)
-	lp := NewLogPoller(o, client.NewSimulatedBackendClient(t, ec, chainID), lggr, 15*time.Second, 2, 3, 2)
+	// Poll period doesn't matter, we intend to call poll and save logs directly in the test.
+	lp := NewLogPoller(o, client.NewSimulatedBackendClient(t, ec, chainID), lggr, 15*time.Second, finalityDepth, backfillBatchSize, rpcBatchSize)
```

Comment:
Would it make sense to use `testutils.WaitTimeout(t)` instead of hard-coded `15 *time.Second`?
Having hard-coded timeouts soon or late result in a flakey test.

Reply:
ah this isn't a timeout, its the poll period so I don't think WaitTimeout is what we want. It is possible to construct a test that would flake if you called Start() and depended on it lp never polling on its own, though we don't currently have one. For testing I think its always cleaner to just call the synchronous pollAndSaveLogs, so I just set this to some high value so it shouldn't poll, but I can increase to be sure it doesn't.